components/connector-service/internal/certificates/certificates_test.go

```diff
@@ -473,22 +519,75 @@ func TestCertificateUtility_CheckCSRValues(t *testing.T) {
 	})
 }
 
-func TestCertificateUtility_SignWithCA(t *testing.T) {
+func TestCertificateUtility_CreateCrtChain(t *testing.T) {
+
+	t.Run("should create certificate chain", func(t *testing.T) {
+		// given
+		certificateUtility := NewCertificateUtility()
+
+		caCrt, csr, key := prepareCrtAndKey(certificateUtility)
+
+		// when
+		crtBase64, err := certificateUtility.CreateCrtChain(caCrt, csr, key)
+
+		// then
+		rawCrt, decodeErr := base64.StdEncoding.DecodeString(crtBase64)
+		decodedCrt, singleCertErr := rawCrtTox509Certificates(rawCrt)
+
+		expectedRawCrt, singleCertErr := rawCrtTox509Certificates([]byte(CrtChain))
+
+		require.NoError(t, singleCertErr)
+		require.NoError(t, err)
+		require.NoError(t, decodeErr)
+
+		clientCrtSubject := expectedRawCrt[0].Subject
+
+		expectedSubject := decodedCrt[0].Subject
+
+		assert.Equal(t, clientCrtSubject, expectedSubject)
+	})
+
+	t.Run("certificate validity days should equal 90", func(t *testing.T) {
+		// given
+		certificateUtility := NewCertificateUtility()
+
+		caCrt, csr, key := prepareCrtAndKey(certificateUtility)
+
+		// when
+		crtBase64, err := certificateUtility.CreateCrtChain(caCrt, csr, key)
+
+		// then
+		rawCrt, decodeErr := base64.StdEncoding.DecodeString(crtBase64)
+		decodedCrt, singleCertErr := rawCrtTox509Certificates(rawCrt)
+
+		require.NoError(t, singleCertErr)
```

Comment:
This error checks should be in different order. If there will be `err` not nil, then following command will fail. Please verify it with team.

Reply:
It should be tested like that:
```
when
then

when
then

when
then
```

Reply:
Pushed.